pkg/operator/ceph/csi/spec.go

```diff
@@ -711,23 +759,27 @@ func (r *ReconcileCSI) stopDrivers(ver *version.Info) error {
 	return nil
 }
 
-func (r *ReconcileCSI) deleteCSIDriverResources(ver *version.Info, daemonset, deployment, service, driverName string) error {
-	csiDriverobj := v1CsiDriver{}
-	err := k8sutil.DeleteDaemonset(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, daemonset)
-	if err != nil {
-		return errors.Wrapf(err, "failed to delete the %q", daemonset)
+func (r *ReconcileCSI) deleteCSIDriverResources(ver *version.Info, daemonset, deployment, service, driverName string, ownerInfo *k8sutil.OwnerInfo) error {
+	if r.daemonOwnedByRook(ownerInfo, daemonset) {
+		err := k8sutil.DeleteDaemonset(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, daemonset)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete the %q", daemonset)
+		}
 	}
 
-	err = k8sutil.DeleteDeployment(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, deployment)
-	if err != nil {
-		return errors.Wrapf(err, "failed to delete the %q", deployment)
+	if r.provisionerOwnedByRook(ownerInfo, deployment) {
+		err := k8sutil.DeleteDeployment(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, deployment)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete the %q", deployment)
+		}
 	}
 
-	err = k8sutil.DeleteService(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, service)
+	err := k8sutil.DeleteService(r.opManagerContext, r.context.Clientset, r.opConfig.OperatorNamespace, service)
 	if err != nil {
 		return errors.Wrapf(err, "failed to delete the %q", service)
 	}
 
+	csiDriverobj := v1CsiDriver{}
```

Comment:
@Madhu-1 btw why we need this empty object?
removing this rook build fails with 
```
=== go vet
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go vet  -tags '' github.com/rook/rook/cmd/... github.com/rook/rook/pkg/... github.com/rook/rook/tests/integration
# github.com/rook/rook/pkg/operator/ceph/csi
pkg/operator/ceph/csi/spec.go:783:21: csiDriverobj.deleteCSIDriverInfo undefined (type v1CsiDriver has no field or method deleteCSIDriverInfo)
# github.com/rook/rook/pkg/operator/ceph/csi

```